### PR TITLE
Allow job-creation operations to specify a project ID

### LINKED ISSUE #1237: Allow project ID to be specified for operations that create jobs
This should default to the project ID of the client, but could be overridden in the options for each call (CreateCopyJobOptions, QueryOptions etc).

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #1237.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/QueryOptions.cs [hunk 1]
```diff
@@ -100,6 +100,12 @@ public sealed class QueryOptions
         /// </summary>
         public bool? DryRun { get; set; }
 
+        /// <summary>
+        /// The ID of the project in which to create the job. If this is not set,
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/QueryOptions.cs]: Probably not "create the job"

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadAvroOptions.cs [hunk 2]
```diff
@@ -41,6 +41,12 @@ public sealed class UploadAvroOptions
         public CreateDisposition? CreateDisposition { get; set; }
 
         /// <summary>
+        /// The ID of the project in which to create the job. If this is not set,
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadAvroOptions.cs]: Probably not "create the job"

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadCsvOptions.cs [hunk 3]
```diff
@@ -87,6 +87,12 @@ public sealed class UploadCsvOptions
         /// </summary>
         public string NullMarker { get; set; }
 
+        /// <summary>
+        /// The ID of the project in which to create the job. If this is not set,
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadCsvOptions.cs]: Probably not "create the job"

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs [hunk 4]
```diff
@@ -52,6 +52,11 @@ public sealed class UploadJsonOptions
         /// </summary>
         public bool? Autodetect { get; set; }
 
+        /// <summary>
+        /// The ID of the project in which to create the job. If this is not set,
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs]: Probably not "create the job"

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/QueryOptions.cs (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/QueryOptions.cs]: Why not? All these operations create a job, and the job is owned by a project. (So I might create a job in my project, even though it's querying public data.) I suspect in theory it's possible for three projects to be involved: "Create a job in project A to copy a table from project B to project C." (So long as the credentials allow that, of course.)

Any suggested wording that would get that idea across more clearly?

**Pull request merged**